### PR TITLE
Authentication implementation

### PR DESCRIPTION
**Pull request by DizzyMelo**

### CODE REVIEW

File: src/config/app.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// Make environment variables available in application
+dotenv.config({ path: path.resolve(__dirname, '../../.env') });
+
+export default {
+  port: process.env.PORT || '3000',
+  clientId: process.env.CLIENT_ID || '',
+  clientSecret: process.env.SECRET || '',
+  redirectUrl: process.env.REDIRECT_URL || '',
+  authorizationHost: process.env.AUTHORIZATION_HOST || '',
+};
```
Comment (eolsenbyui) [inline src/config/app.ts:13]: src/config/app.ts seems to add a layer of complexity that does not seem to be necessary.  Yes, it matches some of the examples we've seen but I don't see that it really adds anything.  The `dotenv` component already encapsulates the configuration values.

At this point, please do not remove it if it hasn't broken anything.

File: src/middleware/loadUser.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { Request, Response, NextFunction } from 'express';
+import config from '../config/app';
+import User from '../models/users';
+
+interface IAuthResponse {
```
Comment (eolsenbyui) [inline src/middleware/loadUser.ts:5]: Should IAuthResponse extend the IUser interface?   I think it should but am willing to listen to arguments to the contrary.

**Review by eolsenbyui — APPROVED** [review]
Please see other comments.

File: src/config/app.ts (hunk 1 above)
Comment (DizzyMelo) [reply src/config/app.ts:13]: It hasn't broken anything, so I think we're not gonna have any problems implementing like this.

File: src/middleware/loadUser.ts (hunk 2 above)
Comment (DizzyMelo) [reply src/middleware/loadUser.ts:5]: I'm not sure, but I think that the IAuthResponse holds elements that are not part of the IUser interface as well as the IUser interface has elements that IAuthResponse doesnt. Even though I am a big fan of inheritance, it doesnt seem like IUser interface is a good candidate to be a base class for IAuthResponse.

Comment (eolsenbyui) [reply src/middleware/loadUser.ts:5]: Shouldn't authenticated users have a one-to-one association with the users in our database?   The users in our database use the IUser interface?    Do you know of a better way to make the association?

These are design questions.   I don't pretend to have all the answers but I do believe that asking the questions will help us to find the right answers.

Comment (eolsenbyui) [reply src/middleware/loadUser.ts:5]: I may be thinking about this backwards.   Go ahead and merge your changes if you feel confident that they won't break anything.

**Pull request merged**